packages/twenty-front/src/modules/auth/hooks/useAuth.ts

```diff
@@ -101,6 +103,18 @@ export const useAuth = () => {
       }
       const workspace = user.defaultWorkspace ?? null;
       setCurrentWorkspace(workspace);
+      if (isDefined(verifyResult.data?.verify.user.workspaces)) {
+        const validWorkspaces =
```

Comment:
This code appears twice and is not very elegant 

Can't you do something shorter?
Like for example (not tested)

```
const workspaces : Workspaces[] = verifyResult.data?.verify.user.workspaces
  ?.filter(({ workspace }) => workspace)
  // then if you need to apply transfo use .map(({ workspace }) => workspace);
```